src/cdk-experimental/menu/menu-item-trigger.ts

```diff
@@ -92,6 +92,24 @@ export class CdkMenuItemTrigger implements OnDestroy {
 
     this._overlayRef = this._overlay.create(this._getOverlayConfig());
     this._overlayRef.attach(this._getPortal());
+
+    this._configureMenu();
+    this._subscribeToClosingActions();
+  }
+
+  /** Wire up the subscription which can cause the menu to close. */
+  private _subscribeToClosingActions() {
+    if (this._overlayRef) {
+      this._overlayRef.outsidePointerEvents().subscribe(() => this.isMenuOpen() && this.toggle());
+    }
+  }
+
+  /** Set the elements array on the open menu and push the open menu to the array. */
+  private _configureMenu() {
+    if (this._menuPanel && this._menuPanel._menu) {
+      this._parentMenu._openMenuElements.push(...this._menuPanel._menu._openMenuElements);
+      this._menuPanel._menu._openMenuElements = this._parentMenu._openMenuElements;
```

Comment:
It seems like reference identity plays a role here. That feels a little subtle and fragile. Could you achieve the same result by pushing elements around, but not having multiple elements share a single menu element array?